commands/operator-sdk/cmd/up/local.go

```diff
@@ -100,12 +104,13 @@ func mustKubeConfig() {
 }
 
 func upLocal() {
-	args := []string{"run", filepath.Join(scaffold.ManagerDir, scaffold.CmdFile)}
+	linkerOpts := fmt.Sprintf("-ldflags \"%s\"", ldFlags)
+	args := []string{"go", "run", linkerOpts, filepath.Join(scaffold.ManagerDir, scaffold.CmdFile)}
```

Comment:
You're seeing expansion issues because `args` in `exec.Command(name, args)` are split by list element and not spaces, ex. `exec.Command("go", []string{"run", "-ldflags \"-X version.Version=1.0.0\""}...)` is not the same as `exec.Command("go", []string{"run", "-ldflags", "-X version.Version=1.0.0"}...)`.

The following works for me:
```Go
func upLocal() {
    	args := []string{"run"}
	if ldFlags != "" {
		args = append(args, []string{"-ldflags", ldFlags}...)
	}
	args = append(args, filepath.Join(scaffold.ManagerDir, scaffold.CmdFile))
	if operatorFlags != "" {
		args = append(args, strings.Split(operatorFlags, " ")...)
	}
	dc := exec.Command("go", args...)
        ....
}
```
The `--shell` flag is therefore not needed.